File: onebot/src/main/java/cn/evole/onebot/sdk/event/notice/misc/OtherClientStatusNoticeEvent.java
package cn.evole.onebot.sdk.event.notice.misc;

import cn.evole.onebot.sdk.event.notice.NoticeEvent;
import com.google.gson.annotations.SerializedName;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;
import lombok.experimental.SuperBuilder;

/**
 * Created on 2022/7/8.
 *
 * @author cnlimiter
 */
@Data
@NoArgsConstructor
@AllArgsConstructor
@SuperBuilder(toBuilder = true)
@EqualsAndHashCode(callSuper = true)
public class OtherClientStatusNoticeEvent extends NoticeEvent {

    @SerializedName("online")
    public boolean online;

    @SerializedName("client")
    private Clients client;

    /**
     * 设备对象
     */
    @Data
    public static class Clients {

        @SerializedName("app_id")
        public long appId;

        @SerializedName("platform")
        public String platform;

        @SerializedName("device_name")
        public String deviceName;

        @SerializedName("device_kind")
        public String deviceKind;

    }

}

File: onebot/src/main/kotlin/sdk/event/notice/misc/OtherClientStatusNoticeEvent.kt
package cn.evolvefield.onebot.sdk.event.notice.misc

import cn.evolvefield.onebot.sdk.event.notice.NoticeEvent
import com.google.gson.annotations.SerializedName

class OtherClientStatusNoticeEvent : NoticeEvent() {
    @SerializedName("online")
    var online = false
    @SerializedName("client")
    var client: Clients? = null

    /**
     * 设备对象
     */
    class Clients {
        @SerializedName("app_id")
        var appId = 0
        @SerializedName("platform")
        var platform = ""
        @SerializedName("device_name")
        var deviceName = ""
        @SerializedName("device_kind")
        var deviceKind = ""
    }
}
